tensorflow/compiler/xla/service/hlo_memory_scheduler.h

```diff
@@ -32,6 +32,12 @@ limitations under the License.
 
 namespace xla {
 
+// Postprocessor of the HloInstructionSequence. This is an opt-in postprocessing
+// function to MemorySchedulerAlgorithm to enforce certain hlo schedule
+// constraints desired for custom-calls.
+typedef std::function<HloInstructionSequence(const HloInstructionSequence&)>
```

Comment:
Is it more composable if we do postprocessing as another pass after memory scheduling?

Reply:
+1

Reply:
The scheduler tries to compare the mem usage of each schedule and pick the lowest one, so ideally postprocessing would happen in this file after the scheduling algorithm, but before the mem usage calculation.

Reply:
Right, the main motivation to implement in the current way (vs. a separate pass) is to account for potential memory size changes caused by the postprocessors. In this way, the change is accounted into the memory usage calculation.

Reply:
@yunxing @dimvar Further thoughts?